### ISSUE TITLE
Changeset generation broken since 4.28

### ISSUE THREAD

**Issue by Floppy012** [liquibase/liquibase#6148]
### Search first

- [X] I searched and no similar issues were found

### Description

Since liquibase 4.28 it seems like that changes are still detected but don't make it to the changeset when using the `liquibase:diff` goal of the maven plugin. This only seems to be the case for when stuff is inserted into the database. When I ran the command on my personal project it did create a changeset containing only drop statements but no create/alter statements.

### Steps To Reproduce

See this minimal reproduction repository: https://github.com/floppy012-repros/liquibase

1. Clone the repo
2. Start the test database `docker compose up -d`
3. `src/resources/db/migrations` is empty
4. Run `mvn clean compile liquibase:diff`
5. Observe log output contains the table and column but also states "No changesets to add to the changelog output."
```
[INFO] Found table test
[INFO] Found table test
[INFO] Found table test
[INFO] Found primary key testPK
[INFO] Found column id bigint
[INFO] Creating snapshot
[INFO] Diff command completed
[INFO] BEST PRACTICE: The changelog generated by diffChangeLog/generateChangeLog should be inspected for correctness and completeness before being deployed. Some database objects and their dependencies cannot be represented automatically, and they may need to be manually updated before being deployed.
[INFO] changeSets count: 0
[INFO] No changesets to add to the changelog output.
```
6. Open `pom.xml` and set `liquibase.version` property to `4.27.0`
7. Run `mvn clean compile liquibase:diff`
8. Observe log output contains the table and column and now also creates the changelog file
```
[INFO] Creating snapshot
[INFO] Found table test
[INFO] Found table test
[INFO] Found table test
[INFO] Found primary key testPK
[INFO] Found column id bigint
[INFO] Creating snapshot
[INFO] Diff command completed
[INFO] BEST PRACTICE: The changelog generated by diffChangeLog/generateChangeLog should be inspected for correctness and completeness before being deployed. Some database objects and their dependencies cannot be represented automatically, and they may need to be manually updated before being deployed.
[INFO] changeSets count: 1
[INFO] src/main/resources/db/migrations/test.xml does not exist, creating and adding 1 changesets.
```
9. Observe the created changelog file in `src/main/resources/db/migrations/test.xml`
10. Tear down the test database `docker compose down -v`

### Expected/Desired Behavior

Liquibase 4.28 and onwards should create a changelog file like 4.27 did.

### Liquibase Version

4.28, 4.29

### Database Vendor & Version

PostgreSQL 16.3.0

### Liquibase Integration

maven

### Liquibase Extensions

liquibase-hibernate6

### OS and/or Infrastructure Type/Provider

Ubuntu 22.04 LTS

### Additional Context

Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)
Maven home: /opt/applications/maven
Java version: 21.0.3, vendor: Ubuntu, runtime: /usr/lib/jvm/java-21-openjdk-amd64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.15.0-112-generic", arch: "amd64", family: "unix"

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by keerthu27** [comment]
Facing same issue when upgrading to Liquibase 4.28.
Its not creating changelog for new fields

**Comment by lorenzbaier** [comment]
this issue is related to https://github.com/liquibase/liquibase-hibernate/issues/708

it breaks because the `Catalog` type is filtered out now because `HibernateDatabase` does not support it, but previously has `setDefault(true)` on the Catalog anyway which does not work anymore because the SnapshotGenerator for it isn't called anymore

I don't know whether this must be fixed in liquibase or liquibase-hibernate but I do think `supportsCatalog` needs to be taken into a count at https://github.com/liquibase/liquibase/blob/e77dcf9db6b1d8f336accc960f8a9c521f2a0e54/liquibase-standard/src/main/java/liquibase/CatalogAndSchema.java#L32 right now it only looks at the comparison database but not the reference...

**Comment by tati-qalified** [comment]
Hi @Floppy012, @keerthu27 and @lorenzbaier, thank you all for reporting this issue. 
Would any of you be willing to submit a PR to propose a fix? Our development team is available to answer any questions you might have.

Let me know!
Thank you,
Tatiana

**Comment by Guschtel** [comment]
I commented a workaround in https://github.com/liquibase/liquibase-hibernate/issues/708#issuecomment-2356362615
Not sure about the implications though, maybe you @tati-qalified can comment on this?

**Comment by Guschtel** [comment]
I think this issue is only related to liquibase-hibernate. not liquibase itself. At lease it can easily be fixed there.

**Comment by sruthiganesh-HC** [comment]
@tati-qalified Any update on this issue yet? This is a major blocker and we cannot downgrade due to recent bug fixes as well.

**Comment by Guschtel** [comment]
@sruthiganesh-HC try the latest nightly build
See https://github.com/liquibase/liquibase-hibernate/commit/cca484af4f19fa4b197bd55f707262a3631df785 that should fix this issue.

**Comment by filipelautert** [comment]
This has been fixed by https://github.com/liquibase/liquibase-hibernate/pull/717, and will be released with Liquibase 4.30.0 later this month.
